lib/state/data/reducer.ts

```diff
@@ -443,6 +443,17 @@ export const tags: A.Reducer<Map<T.TagHash, T.Tag>> = (
       return next;
     }
 
+    case 'RESTORE_NOTE_REVISION': {
+      const next = new Map(state);
+      action.note.tags.map(t).forEach((tagHash) => {
+        if (!next.has(tagHash)) {
+          next.set(tagHash, { name: tagNameOf(tagHash) });
```

Comment:
we have to be careful here. I guess I wrote `tagNameOf` and used it above, but it doesn't give us the casing of the tag. we would want to use the tag name for the `name` property and not the un-hashed value.

```ts
action.ntoe.tags.forEach(tagName => {
    const tagHash = t(tagName);
    if (!next.has(tagHash)) {
        next.set(tagHash, {name: tagName});
    }
})
```

Reply:
Oh good catch, I thought it was safe to use the un-hashed value based on the previous code so I'm going to update this part.

Reply:
Fixed in this [commit](https://github.com/Automattic/simplenote-electron/pull/2817/commits/4c3287a2dd0984c41c0fd87b729ed6693a8d384f).